Unexpected argument when using `-Dno_number_autocast`

This might be sort of an edge case, but we have to use the `-Dno_number_autocast` because without it, Crystal will freely interchange Floats and Ints in typed arguments. For us, this caused some REALLY bad issues. 

When upgrading to Mosquito master branch to test things, everything fails. To recreate this, just update

https://github.com/mosquito-cr/mosquito/blob/0107dba08b18670832423f8e8d725b9b4f35db81/Makefile#L10-L11

and add in `-Dno_number_autocast` and you'll get this error when running `make test`

```
In test/mosquito/job_run/storage_test.cr:39:13

 39 | job_run.delete
              ^-----
Error: instantiating 'Mosquito::JobRun#delete()'


In src/mosquito/job_run.cr:65:5

 65 | def delete(in ttl = 0)
      ^-----
Error: instantiating 'delete(Int32)'


In src/mosquito/job_run.cr:66:43

 66 | Mosquito.backend.delete config_key, ttl
                                          ^--
Error: expected argument #2 to 'Mosquito::TestBackend.delete' to be Int64 or Time::Span, not Int32

Overloads are:
 - Mosquito::TestBackend.delete(key : String, in ttl : Int64 = 0)
 - Mosquito::TestBackend.delete(key : String, in ttl : Time::Span)
make: *** [Makefile:11: test] Error 1
```

Crystal version: 1.12.1
Mosquito Shard version: [master branch](https://github.com/mosquito-cr/mosquito/commit/0107dba08b18670832423f8e8d725b9b4f35db81)

--- a/src/mosquito/job_run.cr
+++ b/src/mosquito/job_run.cr
@@ -62,8 +62,8 @@ module Mosquito
 
     # Deletes this job_run from the backend.
     # Optionally, after a delay in seconds (handled by the backend).
-    def delete(in ttl = 0)
-      Mosquito.backend.delete config_key, ttl
+    def delete(in ttl : Int = 0)
+      Mosquito.backend.delete config_key, ttl.to_i64
     end
 
     # Builds a Job instance from this job_run. Populates the job with config from

